localstack/http/client.py

```diff
@@ -49,11 +67,14 @@ def request(self, request: Request) -> Response:
             headers=Headers(dict(response.headers)),
         )
         if request.method == "HEAD":
-            # for head we have to keep the original content-length, but it will be re-calcualated when creating
+            # for HEAD  requests we have to keep the original content-length, but it will be re-calculated when creating
             # the final_response object
             final_response.content_length = response.headers.get("Content-Length", 0)
         return final_response
 
+    def close(self):
+        self.session.close()
```

Comment:
nit: If a session has been set in the `__init__`, should it really be closed here? In my opinion, the lifecycle of an externally given session should not be handled in here.

Reply:
while i generally agree, i think it's good to stay consistent with established patterns.

it is common throughout IO implementations in different platforms (including Java, and all others I know), that, if you pass an underlying resource to some higher-level abstraction that closable, you're yielding control over that resource to the abstraction. for instance, in this piece of code:

```python
from io import BufferedReader

my_raw_stream = ...

with BufferedReader(my_raw_stream) as reader:
    reader.read()
```

the BufferedReader will call `close()` on `my_raw_stream`.

even though the case may be a bit different, i would treat it the same way. i agree that the drawback is that you cannot re-use session objects across clients. but if you really want to do that, you can always patch the `close` method to a no-op.